### DISCUSSION TITLE
Story Book not populating illustrations

### DISCUSSION THREAD

**Discussion by mkdk2812** [AntlerVC/firetable#1453] [Support / Q&A] [answered]
I have a working Story Book template, but it is not generating images.  The story and pages will all populate but the images never return.  I checked the logs and there are no errors.  It simply sits at "generating illustrations".  This is the 3rd time I've rebuilt the project with the same results.  Please advise.
<img width="936" alt="Screenshot 2023-10-29 at 10 27 42 AM" src="https://github.com/rowyio/rowy/assets/149261977/501ac5f7-26a8-4e4f-b045-9cdf1e04baad">

**Comment by Chadha93** [comment]
@mkdk2812  Based on the information provided, it seems that there might be an issue with the image generation process in Story Book. Here are a few steps you can take to troubleshoot and resolve the issue:

1. Check the image paths: Make sure that the image paths provided in your storybook configuration are correct. Double-check the file names and folder structures to ensure that the images are in the correct location.

2. Verify image formats: Confirm that the images you are using are in a supported format (e.g., JPEG, PNG). Story Book may not be able to generate illustrations if the image format is not supported.

3. Check image loading: Ensure that the images can be loaded in a regular HTML context. Try opening the image URLs directly in a browser to see if they load correctly. If the images are not loading, it could indicate a problem with the image files themselves.

4. Verify image generation setup: If you have customized the image generation setup in Story Book, make sure that it is correctly configured. Check the Story Book documentation or any custom configurations you have made to ensure that the image generation process is set up correctly.

5. Check for any errors or warnings: Even though you mentioned that there are no errors in the logs, it is worth double-checking for any warnings or informational messages related to the image generation process. These messages might provide additional insights into the issue.

If none of the above steps resolve the issue, it might be helpful to provide more information about your Story Book setup (derivative function code etc.), any custom configurations or plugins you are using, and any relevant code snippets.

**Reply by mkdk2812** [reply]
I'm still learning how the script works, but I am basically using the default script that came with the template.  Below is the script as it is now:

const engineId = 'stable-diffusion-xl-beta-v2-2-2'
const apiHost = 'https://api.stability.ai'
const derivative: Derivative = async ({ row, ref, db, storage, auth, logging,tableSchema }) => {
  const apiKey = await rowy.secrets.get(tableSchema.tableEnv.stabilityAI)
  logging.log('Running cover image')
  if (!apiKey) throw new Error('Missing Stability API key.')
  ref.update({status:'drawing'})
  const prompt = row.writingStyle? `an illustration of ${row.response.cover.illustration} for a ${row.writingStyle} book`:`an illustration of ${row.response.cover.illustration}`
  const response = await fetch(
    `${apiHost}/v1/generation/${engineId}/text-to-image`,
    {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
        Authorization: `Bearer ${apiKey}`,
      },
      body: JSON.stringify({
        text_prompts: [
          {
            text: prompt,
          },
        ],
        cfg_scale: 7,
        style_preset:row.stylePreset,
        clip_guidance_preset: 'FAST_BLUE',
        height: 512,
        width: 512,
        samples: 1,
        steps: 50,
      }),
    }
  )

  if (!response.ok) {
    throw new Error(`Non-200 response: ${await response.text()}`)
  }

  interface GenerationResponse {
    artifacts: Array<{
      base64: string
      seed: number
      finishReason: string
    }>
  }

  const responseJSON = (await response.json()) as GenerationResponse
  const imageFile = await rowy.storage.upload.data(  Buffer.from(responseJSON.artifacts[0].base64, 'base64'),{
    folderPath:ref.path +'/coverImage/',
    fileName:'cover-txt-image.png'
  })
  await ref.update({status:'cover generated'})

  return [imageFile];
}

**Comment by BeeBombshell** [comment, marked as answer]
Hey @mkdk2812, can you check if the image is generated when you manually evaluate a single image field?
To manually evaluate a single field, all you need to do is right-click on the cell and click "Evaluate".

<img width="214" alt="image" src="https://github.com/rowyio/rowy/assets/53828745/6f6d406c-cfa2-4d04-af96-6c48a6b7bd3e">

If this doesn't work, try adding a few [logging statements](https://docs.rowy.io/cloud-logs) and check the cloud logs specific to the column for any particular error message.
